### DISCUSSION TITLE
cant not use @can on middelware with resource controller

### DISCUSSION THREAD

**Discussion by cesarMtorres** [spatie/laravel-permission#2157] [Q&A] [answered]
hi there, why i cant not using this way in the controllers resource?

`php
Route::group(['middleware'` => ['auth', 'permission:roles.index|roles.destroy|roles.addPermissionsToRole']], function () {

    Route::resource('roles', RoleController::class);

    Route::controller(RoleController::class)->prefix('roles')->name('roles.')->group(function () {
       ...
       ...
    });
});

this only works by adding it in the __contructor()
or by using $this->authorize('roles.index')

**Comment by angeljqv** [comment, marked as answer]
Don't use can, there is others middleware, read documentation

**Reply by cesarMtorres** [reply]
spatie support resource controller with multiple permission?

**Reply by angeljqv** [reply]
Yes, but it depends of your implementation, read documentation, there is examples (`Authorizable Trait`)

**Reply by cesarMtorres** [reply]
use permission middleware was the answer

**Reply by angeljqv** [reply]
So, mark this as answered
